### DISCUSSION TITLE
ModuleNotFoundError: No module named 'supervision'

### DISCUSSION THREAD

**Discussion by mdawood1520** [lllyasviel/Fooocus#3284] [Q&A] [answered]
An error occurred while installing version 2.5.0. Ask for help

**Comment by mashb1t** [comment]
2.5.0 is not out yet, i assume you mean 2.5.0-rc1.
This version introduces various package updates, which depending on your ISP/firewall/country/VPN may be blocked. Please check and provide the terminal logs.

**Reply by mdawood1520** [reply]
![9605340de343eae076d3d9aa7307488](https://github.com/user-attachments/assets/d33930c5-3569-45ff-919f-c5d9fe521212)
Is this the terminal log you're referring to? Sorry, I'm new

**Reply by mashb1t** [reply]
As you can see at the top of your log there has been an issue with the installation of packages, but it's sadly cut off. Please check this error and resolve it if possible.

**Reply by mdawood1520** [reply]
> As you can see at the top of your log there has been an issue with the installation of packages, but it's sadly cut off. Please check this error and resolve it if possible.
D:\Fooocus_win64_2.5.0rci>.\python_embeded\python.exe -s Fooocus\entry_with_update.py
Already up-to-date
Update succeeded.
[System ARGV] ['Fooocus\\entry_with_update.py']
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.5.0-rc1
Version mismatch for tqdm: Installed version 4.64.1 does not meet requirement tqdm==4.65.0
Installing requirements
Couldn't install requirements.
Command: "D:\Fooocus_win64_2.5.0rci\python_embeded\python.exe" -m pip install -r "requirements_versions.txt" --prefer-binary
Error code: 1
stdout: Requirement already satisfied: torchsde==0.2.5 in d:\fooocus_win64_2.5.0rci\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 1)) (0.2.5)
Requirement already satisfied: einops==0.4.1 in d:\fooocus_win64_2.5.0rci\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 2)) (0.4.1)
Requirement already satisfied: transformers==4.30.2 in d:\fooocus_win64_2.5.0rci\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 3)) (4.30.2)
Requirement already satisfied: safetensors==0.3.1 in d:\fooocus_win64_2.5.0rci\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 4)) (0.3.1)
Requirement already satisfied: accelerate==0.21.0 in d:\fooocus_win64_2.5.0rci\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 5)) (0.21.0)
Requirement already satisfied: pyyaml==6.0 in d:\fooocus_win64_2.5.0rci\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 6)) (6.0)
Requirement already satisfied: Pillow==9.2.0 in d:\fooocus_win64_2.5.0rci\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 7)) (9.2.0)
Requirement already satisfied: scipy==1.9.3 in d:\fooocus_win64_2.5.0rci\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 8)) (1.9.3)
Collecting tqdm==4.65.0 (from -r requirements_versions.txt (line 9))
  Using cached tqdm-4.65.0-py3-none-any.whl.metadata (56 kB)
Requirement already satisfied: psutil==5.9.5 in d:\fooocus_win64_2.5.0rci\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 10)) (5.9.5)
Requirement already satisfied: pytorch_lightning==1.9.4 in d:\fooocus_win64_2.5.0rci\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 11)) (1.9.4)
Requirement already satisfied: omegaconf==2.2.3 in d:\fooocus_win64_2.5.0rci\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 12)) (2.2.3)
Requirement already satisfied: gradio==3.41.2 in d:\fooocus_win64_2.5.0rci\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 13)) (3.41.2)
Requirement already satisfied: pygit2==1.12.2 in d:\fooocus_win64_2.5.0rci\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 14)) (1.12.2)
Requirement already satisfied: opencv-contrib-python==4.8.0.74 in d:\fooocus_win64_2.5.0rci\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 15)) (4.8.0.74)
Requirement already satisfied: httpx==0.24.1 in d:\fooocus_win64_2.5.0rci\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 16)) (0.24.1)
Requirement already satisfied: onnxruntime==1.16.3 in d:\fooocus_win64_2.5.0rci\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 17)) (1.16.3)
Requirement already satisfied: timm==0.9.2 in d:\fooocus_win64_2.5.0rci\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 18)) (0.9.2)
Collecting translators==5.9.2 (from -r requirements_versions.txt (line 19))
  Using cached translators-5.9.2-py3-none-any.whl.metadata (68 kB)
Collecting rembg==2.0.57 (from -r requirements_versions.txt (line 20))
  Using cached rembg-2.0.57-py3-none-any.whl.metadata (16 kB)
Collecting groundingdino-py==0.4.0 (from -r requirements_versions.txt (line 21))
  Using cached groundingdino-py-0.4.0.tar.gz (82 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'

stderr:   error: subprocess-exited-with-error

  python setup.py egg_info did not run successfully.
  exit code: 1

  [6 lines of output]
  Traceback (most recent call last):
    File "<string>", line 2, in <module>
    File "<pip-setuptools-caller>", line 34, in <module>
    File "C:\Users\Pc\AppData\Local\Temp\pip-install-l1abktdh\groundingdino-py_02dd3fcb0fa346c9b98aae9ae1168cb5\setup.py", line 41, in <module>
      readme = readme_file.read()
  UnicodeDecodeError: 'gbk' codec can't decode byte 0xa4 in position 2878: illegal multibyte sequence
  [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

Encountered error while generating package metadata.

See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

CMD Failed requirements: install -r "requirements_versions.txt"
[Cleanup] Attempting to delete content of temp dir C:\Users\Pc\AppData\Local\Temp\fooocus
[Cleanup] Cleanup successful
Total VRAM 8188 MB, total RAM 32610 MB
Set vram state to: NORMAL_VRAM
Always offload VRAM
Device: cuda:0 NVIDIA GeForce RTX 4060 : native
VAE dtype: torch.bfloat16
Traceback (most recent call last):
  File "D:\Fooocus_win64_2.5.0rci\Fooocus\entry_with_update.py", line 46, in <module>
    from launch import *
  File "D:\Fooocus_win64_2.5.0rci\Fooocus\launch.py", line 144, in <module>
    from webui import *
  File "D:\Fooocus_win64_2.5.0rci\Fooocus\webui.py", line 10, in <module>
    import modules.async_worker as worker
  File "D:\Fooocus_win64_2.5.0rci\Fooocus\modules\async_worker.py", line 3, in <module>
    from extras.inpaint_mask import generate_mask_from_image, SAMOptions
  File "D:\Fooocus_win64_2.5.0rci\Fooocus\extras\inpaint_mask.py", line 6, in <module>
    from extras.GroundingDINO.util.inference import default_groundingdino
  File "D:\Fooocus_win64_2.5.0rci\Fooocus\extras\GroundingDINO\util\inference.py", line 9, in <module>
    import supervision as sv
ModuleNotFoundError: No module named 'supervision'

D:\Fooocus_win64_2.5.0rci>pause
This is all the logs after 'run'. Could you please help me solve the problem

**Reply by mashb1t** [reply, marked as answer]
Happy to report that you can find the solution to this in https://github.com/lllyasviel/Fooocus/discussions/3282#discussioncomment-10046423

**Reply by antoniormrzz** [reply]
@mashb1t unfortunately the linked comments are not solving the issue. The main comment isn't detailed enough as to where to put the zip of the package after downloading it, and where to run the actual pip command. Changing the region didn't help much either. Is there a way to fix this without a clean install?

**Reply by mashb1t** [reply]
@antoniormrzz does this video help you to update to Fooocus 2.5.0?
https://www.youtube.com/watch?v=cxjnzTpV4cg

**Reply by antoniormrzz** [reply]
@mashb1t I saw your response that we should follow https://github.com/lllyasviel/Fooocus/discussions/3293 which solved the problem. Can we get a pin of this for other users? Also, some repos add a section at the top of readme with either short instructions on update or a link. 

Btw, this was a huge update, I was waiting for something really big in this repo, and this is definitely it, something that significantly improves Fooocus's capabilities. I want to thank you and other contributors for your work on this update. I haven't given it a try yet, but being able to enhance specific parts of image without constant inpainting or using image editing tools sounds great.

**Reply by mashb1t** [reply]
@antoniormrzz nice hint, pinned in https://github.com/lllyasviel/Fooocus/discussions

**Reply by 0xcyph** [reply]
> @antoniormrzz does this video help you to update to Fooocus 2.5.0? https://www.youtube.com/watch?v=cxjnzTpV4cg

worked for me! thx!

**Comment by Natibliss** [comment]
Yesterday I was running FOOOCUS without a problem, and now I get this error, please help me.

File "/kaggle/working/Fooocus/extras/GroundingDINO/util/inference.py", line 9, in <module>
    import supervision as sv
ModuleNotFoundError: No module named 'supervision'
connect_to localhost port 7865: failed.
connect_to localhost port 7865: failed.

**Reply by mashb1t** [reply]
@Natibliss please follow the "How to update" section in https://github.com/lllyasviel/Fooocus/discussions/3293 and double-check already existing discussions and previous comments about this next time :)

**Reply by mashb1t** [reply]
there even is a video tutorial in https://github.com/lllyasviel/Fooocus/discussions/3293#discussioncomment-10077368

**Reply by Natibliss** [reply]
new error:

File "/opt/conda/lib/python3.10/site-packages/scipy/sparse/linalg/isolve/utils.py", line 7, in <module>
    from scipy.sparse.sputils import asmatrix
ImportError: cannot import name 'asmatrix' from 'scipy.sparse.sputils' (/opt/conda/lib/python3.10/site-packages/scipy/sparse/sputils.py)
connect_to localhost port 7865: failed.

**Reply by mashb1t** [reply]
same solution, please ensure you follow all the steps in the how to update section of the release notes

**Reply by romxluv** [reply]
Actually you can use it if your pc is not connected to internet

**Comment by ravidnr** [comment]
Open your fooocus folder
type cmd in the path-bar
copy and paste the below command and press enter

..\python_embeded\python.exe -m pip install -r .\requirements_versions.txt

exit the cmd prompt, now you can run the fooocus properly

**Reply by Penny13692018** [reply]
"..\python_embeded\python.exe -m pip install -r .\requirements_versions.txt"
How to do the command in Kaggle notebook?
I tried 
!pip install -r requirements_versions.txt
But it still not works

**Reply by mashb1t** [reply]
@Penny13692018 You have to be in the correct path, so please locate the file and run the command afterwards.

**Reply by Penny13692018** [reply]
@mashb1t  I cannot find the right path in Kaggle.
I ran 
> !pip install -r requirements_versions.txt

Under the Fooocus directory of Kaggle since there’s no dir of “ python_embeded” in Kaggle.

It processed but didn’t fix the problem.

**Reply by mashb1t** [reply]
@Penny13692018 There is no official Kaggle support, so maybe somebody else using Kaggle can help you. I'd propose to open a new discussion with Kaggle in the title.

**Reply by Penny13692018** [reply]
How to install and run 2.4.3 version in Kaggle? 
For this notebook it can run 2.4.3 properly, why?
https://www.kaggle.com/code/conforterworm/fooocus

But for this notebook,I can not use version under 2.5.0 and it breaks down.
https://www.kaggle.com/code/conforterworm/fooocus-anime-test

**Reply by Hajmer777** [reply]
python -m pip install -r .\requirements_versions.txt
pip -m pip install -r .\requirements_versions.txt
